### ISSUE TITLE
Blazor - (client side) Route parameter with dot (.) not working

### ISSUE THREAD

**Issue by royzhang666** [dotnet/aspnetcore#13247]
### Describe the bug
The client side blazor (wasm) component cannot match the route parameter with dot (.) when you open the link the first time, instead it will return 404 error. It worked when you navigate from other blazor components.

### To Reproduce
1. Using this version of ASP.NET Core 3.0.0-preview8.19405.7
2. Run this blazor component

```
@page "/hellow/{Name}"

@Name

@code
{
    [Parameter]
    public string Name { get; set; }
}
```

3. With these arguments '/hellow/hello.world'
4. See error: 404

### Expected behavior
the page containing name **hello.world**

### Screenshots

<img width="760" alt="Screen Shot 2019-08-19 at 3 54 53 PM" src="https://user-images.githubusercontent.com/17616236/63248951-07e9e800-c29b-11e9-8162-39c3a86dd8e5.png">

**Comment by rynowak** [comment]
The route that's [added by default](https://github.com/aspnet/AspNetCore/blob/69a703748026b4890db2e6ce683c05f5f3fffe1c/src/Http/Routing/src/Builder/FallbackEndpointRouteBuilderExtensions.cs#L18) when you use `MapFallbackToClientSideBlazor` (in Startup) intentionally does not match URLs that look like filenames (contains a `.` surrounded by other characters in the last segment). This is on purpose so that your app will give a proper 404 for requests that don't match a static file.

If you want to change this, specify your own route pattern - you could use `{**path}` and then it will happily match files if that's what you want.

**Comment by mkArtakMSFT** [comment]
We'll try to cover this with docs.

**Comment by pranavkm** [comment]
Doing this for you.

**Comment by pranavkm** [comment]
This requires more than documentation. When using dev-server, you are limited to using the values configured by the dev-server. In this case, the server uses a fixed fallback route:

https://github.com/aspnet/AspNetCore/blob/e0f95cfa6feec402ea34a605965f356720006513/src/Components/Blazor/DevServer/src/Server/Startup.cs#L58

Moving this to the next milestone to figure out how to tackle this.

**Comment by mkArtakMSFT** [comment]
@pranavkm is there anything left to be done here>?

**Comment by Silence-Among-Crows** [comment]
Hi there, I ended up with what I think may be a similar issue.

I am using Client side Blazor and have stumbled upon a tricky section regards to routing with strings and dots (.). I wish to route from an admin control page to a page with the following route: `@page "/ManageGradingExamResults/{StudentEmail}"`. I tested such with this request: https://localhost/ManageGradingExamResults/1234@high.school.nz, but I get a not found. If I change the razor route to `@page "/ManageGradingExamResults/{*StudentEmail}"` I end up with the following exception: `System.InvalidOperationException: Invalid template 'ManageGradingExamResults/{*StudentEmail}'. The character '*' in parameter segment '{*StudentEmail}' is not allowed..` I tried that due to the similarities in cshtml pages.

I have found that I can route with integers but have had no luck with strings. I also have come across this Microsoft doc explaining route params and suggesting the ** in my page route. This allows me into my page however i then need to use the ? before my email in the request, the page does not load with my data as I can see it did not pull the parameter into my variable. Any suggestions or help with routing in client side Blazor would be greatly appreciated!

_Request Code:_

`NavigationManager.NavigateTo($"/ManageGradingExamResults/?{student.Email}");`

_Request URL:_

`https://localhost/ManageGradingExamResults/?1234@high.school.nz`

_Razor page route:_

```
@page "/**ManageGradingExamResults/{StudentEmail}"
@page "/ManageGradingExamResults"
```

_My Variable:_

```
@code
{
    [Parameter]
    public string StudentEmail { get; set; }
...
```
